plugin/src/main/java/org/jenkinsci/plugins/structs/describable/DescribableModel.java

```diff
@@ -284,13 +309,24 @@ public T instantiate(Map<String,?> arguments) throws Exception {
             arguments = Collections.singletonMap(rp.getName(),arguments.get(ANONYMOUS_KEY));
         }
 
+        Set<String> erroneous =  new TreeSet<>(arguments.keySet());
+        erroneous.removeAll(parameters.keySet());
+        if (erroneous.size() > 0) {
+            String msg = "WARNING: Unknown parameter(s) found for class type '" + this.type.getName() + "': " + String.join(",", erroneous);
+            if (Main.isUnitTest) {
```

Comment:
nit: Generally a good idea to create a `public static` (non-final) variable to control the behavior so that if necessary, it can be turned on or off via the script console in Jenkins.

Reply:
Ok, sorry, I was unclear, I meant use the variable to _replace_ of the direct inline use of `Main.isUnitTest`: `if (STRICT_PARAMETER_CHECKING) {` here, and then have `STRICT_PARAMETER_CHECKING = Main.isUnitTest`. That way someone can control whether to get exceptions or just warnings in the log with that parameter (and you can set it to false in a test to have a test that checks logging behavior). I don't think there is a need for control over whether to log or not if you are not throwing an exception like you added below, I would revert that part of the change.  I would also add `@Restricted(NoExternalUse.class)` to the field so no other plugin depends on it.


 [Here](https://github.com/jenkinsci/workflow-cps-plugin/blob/2974e6cfffe6a8e94272047ba504923f2823aed3/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsVmExecutorService.java#L143) is an example of a similar variable in workflow-cps and [here](https://github.com/jenkinsci/workflow-cps-plugin/blob/2974e6cfffe6a8e94272047ba504923f2823aed3/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsVmExecutorService.java#L116-L124) is where it is used to do the same kind of thing as what you are doing here (the variable is package-private in that case, so only useful in testing, but I think public is fine "just in case").